File: backend/src/main/java/net/archasmiel/skufapi/api/exception/ErrorResponse.java
package net.archasmiel.skufapi.api.exception;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import net.archasmiel.skufapi.api.response.ApiResponse;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Response for error")
public class ErrorResponse implements ApiResponse {

    @Schema(description = "Error", example = "GOOGLE_AUTH_FAILED")
    private String error;

    @Schema(description = "Message", example = "This error means...")
    private String message;
}

File: backend/src/main/kotlin/net/archasmiel/skufapi/api/exception/ErrorResponse.kt
package net.archasmiel.skufapi.api.exception

import io.swagger.v3.oas.annotations.media.Schema
import net.archasmiel.skufapi.api.response.ApiResponse

@Schema(description = "Response for error")
data class ErrorResponse(
    @field:Schema(description = "Error code", example = "GOOGLE_AUTH_FAILED")
    val code: String,

    @field:Schema(description = "Message string", example = "This error means...")
    val message: String
) : ApiResponse